iosHyperskillApp/iosHyperskillApp/Sources/Modules/Debug/DebugViewModel.swift

```diff
@@ -0,0 +1,55 @@
+import Darwin
+import shared
+import SwiftUI
+
+final class DebugViewModel: FeatureViewModel<
+  DebugFeatureViewState,
+  DebugFeatureMessage,
+  DebugFeatureActionViewAction
+> {
+    private let notificationsService: NotificationsService
+
+    var stateKs: DebugFeatureViewStateKs { .init(state) }
+
+    init(
+        notificationsService: NotificationsService,
+        feature: Presentation_reduxFeature
+    ) {
+        self.notificationsService = notificationsService
+        super.init(feature: feature)
+    }
+
+    override func shouldNotifyStateDidChange(
+        oldState: DebugFeatureViewState,
+        newState: DebugFeatureViewState
+    ) -> Bool {
+        DebugFeatureViewStateKs(oldState) != DebugFeatureViewStateKs(newState)
+    }
+
+    func doLoadDebugSettings(forceUpdate: Bool = false) {
+        onNewMessage(DebugFeatureMessageInitialize(forceUpdate: forceUpdate))
+    }
+
+    func doSelectEndpointConfig(_ endpointConfig: EndpointConfigType) {
+        onNewMessage(DebugFeatureMessageSelectEndpointConfig(endpointConfig: endpointConfig))
+    }
+
+    func doStepNavigationInputChange(text: String) {
+        onNewMessage(DebugFeatureMessageStepNavigationInputTextChanged(text: text))
+    }
+
+    func doStepNavigationOpenStep() {
+        onNewMessage(DebugFeatureMessageStepNavigationOpenClicked())
+    }
+
+    func doApplySettings() {
+        onNewMessage(DebugFeatureMessageApplySettingsClicked())
+    }
+
+    func doRestartApplication() {
+        Task {
+            await notificationsService.scheduleRestartApplicationLocalNotification()
+            exit(0)
```

Comment:
Do I understand correctly that we are completing the main thread here, because this function is launched by pressing the button and therefore the Task is executed in main thread?

Reply:
The `exit(0)` function is used to terminate the program immediately. The argument `0` passed to the `exit` function is the exit status of the program, which is an integer that indicates the success or failure of the program's execution. An exit status of `0` typically indicates that the program executed successfully, while non-zero exit statuses indicate that some error or exception occurred during the program's execution.

This function provides `Darwin` framework, you can see its import at the top of the file. The Darwin framework serves as a foundation for the Apple operating system and provides the basic building blocks for the development of more complex applications and services.